app/controllers/documents_controller.rb

```diff
@@ -59,7 +59,19 @@ def start; end
 
   def new_document
     document = Document.new(document_params.merge(chunking_profile: @chunking_profile))
-    document.filename = params[:file].original_filename
+    if params[:link].present?
+      begin
+        f = Tempfile.create(['web-', '.html'])
+        f.puts HTTParty.get(params[:link])
+        f.rewind
+        document.file = f
+        document.filename = File.basename(f.path)
+      rescue StandardError
+        document.update(state: :errored)
```

Comment:
This should include the exception message in the `error_message` field. The error message will show when the user hovers over the "Error" pill on the document in the UI.